### ISSUE TITLE
stat_histinterval: odd behavior for groups with no variance

### ISSUE THREAD

**Issue by steveharoz** [mjskay/ggdist#32]
If all values in a group are the same (not uncommon for a baseline or control condition), the histogram borks.
Maybe something to do with bin size?

```r
set.seed(1234)
df = tribble(
  ~group, ~value,
  "a",    rep(10, 1000), # no variance here
  "b",    rnorm(1000, mean = -5),
  "c",    rnorm(1000, mean = 10),
) %>%
  unnest(value)

df %>%
  ggplot(aes(y = group, x = value)) +
  stat_histinterval(slab_type = "histogram") +
  theme_ggdist()
```

![image](https://user-images.githubusercontent.com/2257540/98709546-d1f2c100-2382-11eb-82d7-d4d547d7654f.png)

A workaround is to filter out the non-variant group and add it as an annotation.

**Comment by steveharoz** [comment]
The same thing happens with `hist(rep(10, 1000))`, so this probably isn't worth fixing in ggdist.

**Comment by mjskay** [comment]
Hmph that is kinda obnoxious though. I assume it's the default binning for data that's all equal. I have been considering swapping out the binning algorithm, and this might be a good motivation to do it.

Either way you'd probably need to special case this, since constants will have much taller bins and everything else would get scaled down too much. On the other hand this isn't am uncommon use case so having an argument to just drop the slab on constants might be useful?

**Comment by steveharoz** [comment]
Yeah, dropping the slab could work. Although the single bar conveys the point really well even if it's vertically truncated. As you saw elsewhere, I wound up replacing the baseline condition with a `geom_vline()`, so maybe that's a general recommendation for folks with similar data.

Here's a related quirk and feature request:
It'd be great to be able to specify the bin width (not the breaks) for all of the histograms, so it's visually consistent. It would somewhat solve the above problem too.

```
set.seed(1234)
tribble(
  ~group, ~value,
  "b",    rnorm(1000, mean = 6, sd = 3),
  "c",    rnorm(1000, mean = 8, sd = 2),
  "d",    rnorm(1000, mean = 10),
) %>%
  unnest(value)  %>%
  ggplot() +
  aes(y = group, x = value) +
  stat_histinterval(slab_type = "histogram") +
  theme_ggdist()
```

![image](https://user-images.githubusercontent.com/2257540/98964433-14460a80-2509-11eb-951d-efd362424ed0.png)
 
It'd be nice if there were a parameter to specify fixed bin sizes for all histograms. Like `bindwidth = 0.5, binstart=-0.25`. And then all bins would be [-0.25 + 0.5n. 0.25 + 0.5n)

**Comment by mjskay** [comment]
This is now fixed as a consequence of restructuring in #103 and #83 in the `unify-dist` branch. Constant distributions are detected and rendered as point masses throughout the `stat_slabinterval()` family. In this case that results in this:

```r
set.seed(1234)
df = tribble(
  ~group, ~value,
  "a",    rep(10, 1000), # no variance here
  "b",    rnorm(1000, mean = -5),
  "c",    rnorm(1000, mean = 10),
) %>%
  unnest(value)

df %>%
  ggplot(aes(y = group, x = value)) +
  stat_histinterval(slab_type = "histogram") +
  theme_ggdist()
```
![image](https://user-images.githubusercontent.com/6345019/144774550-8f8e6094-5b87-4148-8db7-4ba094cdb344.png)

Which may seem a big odd, but if you enable the outline on the slab you can see what's going on:

```r
df %>%
  ggplot(aes(y = group, x = value)) +
  stat_histinterval(slab_type = "histogram", slab_color = "gray75") +
  theme_ggdist()
```
![image](https://user-images.githubusercontent.com/6345019/144774643-dc2bcee2-ed68-4a00-b47a-39595e6fb972.png)

And the new `expand` parameter can also let you see this in a more ridgeline-like way:

```r
df %>%
  ggplot(aes(y = group, x = value)) +
  stat_histinterval(slab_type = "histogram", slab_color = "gray75", expand = TRUE) +
  theme_ggdist()
```
![image](https://user-images.githubusercontent.com/6345019/144774787-c3eb44d0-799f-4f0c-bb2c-de8a58ae2b45.png)

```
